### PR TITLE
An authentication method is added.

### PR DESCRIPTION
**Pull request by Manda-supraja26**
# Tasks / Issue :construction:
- [ ] #6 

# Solution :heavy_check_mark:
```go
func authenticate() (string, error) {
	apikey, err := fetchAPIKey()
	if err != nil {
		return "", err
	}
	civoToken := os.Getenv("CIVO_TOKEN")
	if civoToken != "" {
		return civoToken, nil
	}
	return apikey, nil

}
```

# Screenshots :framed_picture:

# Note to reviewers :notebook:

### CODE REVIEW

**Comment by dipankardas011** [comment]
@Manda-supraja26 its a breaking change for the modules which utilize this function
also, review the azure pkg as well

File: api/civo/main.go [hunk 1]
```diff
@@ -50,15 +50,29 @@ func Credentials() bool {
...
+	apikey, err := fetchAPIKey()
+	if err != nil {
+		return "", err
+	}
+	civoToken := os.Getenv("CIVO_TOKEN")
+	if civoToken != "" {
+		return civoToken, nil
+	}
+	return apikey, nil
+
```
Comment (dipankardas011) [inline api/civo/main.go:75]: make the CIVO_TOKEN Env check before the fetchAPIKey() also edit the fetchAPIKey no need to make it seperate

@AvineshTripathi should we have 2 seperate function one for env and another which access the token from the `cred.json`?

**Review by dipankardas011 — CHANGES_REQUESTED** [review]
Changes are required

File: api/civo/main.go (hunk 1 above)
Comment (AvineshTripathi) [reply api/civo/main.go:75]: I think we should support only one

Comment (dipankardas011) [reply api/civo/main.go]: Yes

**Comment by Manda-supraja26** [comment]
made changes as requested. If there are any changes, please let me know.

**Comment by dipankardas011** [comment]
@Manda-supraja26 whats the point of keeping the error as the second return value as we are not utilizing the value just suppressing it in all the places may be we don't need that

**Comment by dipankardas011** [comment]
Also make the necessary changes to the azure part as well

**Comment by Manda-supraja26** [comment]
> Also make the necessary changes to the azure part as well

I didn't get what changes I should make to the azure folder.

**Pull request closed without merging**